ICSharpCode.Decompiler/CSharp/Transforms/ReplaceMethodCallsWithOperators.cs

```diff
@@ -106,8 +108,67 @@ void ProcessInvocationExpression(InvocationExpression invocationExpression)
 						invocationExpression.ReplaceWith(new ObjectCreateExpression(context.TypeSystemAstBuilder.ConvertType(method.TypeArguments.First())));
 					}
 					break;
+				case "System.String.Format":
+					if (context.Settings.StringInterpolation && arguments.Length > 1
+						&& arguments[0] is PrimitiveExpression stringExpression && stringExpression.Value is string
+						&& arguments.Skip(1).All(a => !a.DescendantsAndSelf.OfType<PrimitiveExpression>().Any(p => p.Value is string)))
```

Comment:
You only check for normal string literals.
Formatted string literals are probably also forbidden?
What about char literals?
What about the `?:` operator (might be problematic due to having a `:`)?

Reply:
I just tested with char literals and interpolated strings... seems fine to use them.
As for conditional expressions: seems they're only valid when wrapped in parentheses.